packages-ts/starknet-gauntlet/src/commands/base/executeCommand.ts

```diff
@@ -188,13 +188,19 @@ export const makeExecuteCommand = <UI, CI>(config: ExecuteCommandConfig<UI, CI>)
 
     // TODO: This will be required for Multisig
     makeMessage = async (): Promise<Call[]> => {
-      const contract = new Contract(this.contract.abi, this.contractAddress, this.provider.provider)
-      const invocation = contract.populate(
-        config.internalFunction || config.action,
-        this.input.contract as any,
-      )
-
-      return [invocation]
+      // If more than one argument is passed in, then the following assumptions will be made:
+      //
+      //  - All `args` are valid Starknet contract addresses
+      //  - All contract addresses reference contracts with the same ABI
```

Comment:
looks good to me, but how often is it that you will have the same inputs to different contracts?

Reply:
I actually have the same question here 👀 

If we do need to support batching with different inputs, then my understanding is that we'd need to modify the flags for the relevant commands to take in an array of values (e.g. `--address addr1` would now need to be `--address addr1,addr2,..`)

In my discussions with @archseer, he mentioned that this would be a lot of extra work and that we shouldn't change any of the existing flags

Based on this, I think the answer to your original question is that it should happen frequently enough to make this change worth it. Other cases which involve different inputs to different contracts would have to be covered by G++.